core/trie2/triedb/pathdb/buffer.go

```diff
@@ -0,0 +1,75 @@
+package pathdb
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2/trienode"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+	"github.com/NethermindEth/juno/db"
+)
+
+// Stores the pending nodes in memory to be committed later
+type buffer struct {
+	layers uint64
+	limit  uint64
+	nodes  *nodeSet
+}
+
+func newBuffer(limit int, nodes *nodeSet, layer uint64) *buffer {
+	if nodes == nil {
+		nodes = newNodeSet(nil, nil, nil)
+	}
+	return &buffer{
+		limit:  uint64(limit),
+		nodes:  nodes,
+		layers: layer,
+	}
+}
```

Comment:
Why return a pointer and not a value type?

Reply:
If there is only meant to exist only one single instance of this type through out execution?